### PR TITLE
Fix poke-type component plural madness

### PR DESCRIPTION
**Pull request by MajorBreakfast**
Now the `<poke-type>` component simply takes a string property `type`, not an array. Notice, that this PR does not lack any files. This has been partially implemented before. Not sure why it even worked though.

Note: I've removed the type name and type color definitions from the api service as its static data not from the api

### CODE REVIEW

**Comment by WoH** [comment]
I wrapped other static data (ProjectGroups) in a Service. It's a bit of [over-engineering](https://cdn-images-1.medium.com/max/800/1*1ceZ3tCVJ-iRyS7B2THn7Q.png), but we shouldn't mix both.

**Comment by MajorBreakfast** [comment]
Have you merged this already? If not, then I'd like to merge this PR first. Then you can pull and integrate your service. May I merge?

**Comment by WoH** [comment]
The service has been in there for a while.

https://github.com/PokemonGoers/Catch-em-all/blob/develop/ionic2/src/services/project-groups.service.ts

**Comment by MajorBreakfast** [comment]
Go to go?

**Review by WoH — APPROVED** [review]

File: ionic2/src/components/filter-pokemon-tab/filter-pokemon-tab.component.ts [hunk 1]
```diff
@@ -29,7 +29,9 @@ export class FilterPokemonTabComponent implements OnInit {
     pokemonTypes: []
   };
 
-  constructor(private apiService: ApiService, private filterService: FilterService) { }
+  constructor(private apiService: ApiService,
+              private filterService: FilterService,
+              private typeService: TypeService) { }
```
Comment (WoH) [inline ionic2/src/components/filter-pokemon-tab/filter-pokemon-tab.component.ts]: !For future reference only, i'll fix it asap.
I missed something there: This should be public.
Reason being: If you use the service in a template, the AoT compile must be able to access it.

File: ionic2/src/components/type/type.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import { TypeService } from '../../services/type.service';
+
+@Component({
+  selector: 'poke-type',
+  templateUrl: './type.component.html'
+})
+export class TypeComponent {
+  @Input() type: string;
+  constructor(private typeService: TypeService) { }
```
Comment (WoH) [inline ionic2/src/components/type/type.component.ts:11]: !For future reference only, i'll fix it asap.
I missed something there: This should be public.
Reason being: If you use the service in a template, the AoT compile must be able to access it.

Comment (MajorBreakfast) [reply ionic2/src/components/type/type.component.ts:11]: Ah okay. But it did work somehow. Is ahead of time compilation disabled in dev mode?

Comment (WoH) [reply ionic2/src/components/type/type.component.ts:11]: Yes

**Pull request merged**